src/ert/config/ert_config.py

```diff
@@ -780,41 +776,19 @@ def __eq__(self, other: object) -> bool:
         return True
 
     @staticmethod
-    def with_plugins(
-        forward_model_step_classes: list[type[ForwardModelStepPlugin]] | None = None,
-        env_pr_fm_step: dict[str, dict[str, Any]] | None = None,
-    ) -> type[ErtConfig]:
-        from ert.plugins import ErtPluginManager  # noqa: PLC0415
-
-        pm = ErtPluginManager()
-        if forward_model_step_classes is None:
-            forward_model_step_classes = pm.forward_model_steps
-
-        preinstalled_fm_steps: dict[str, ForwardModelStepPlugin] = {}
-        for fm_step_subclass in forward_model_step_classes:
-            # we call without required arguments to
-            # ForwardModelStepPlugin.__init__ as
-            # we expect the subclass to override __init__
-            # and provide those arguments
-            fm_step = fm_step_subclass()  # type: ignore
-            preinstalled_fm_steps[fm_step.name] = fm_step
-
-        if env_pr_fm_step is None:
-            env_pr_fm_step_ = uppercase_subkeys_and_stringify_subvalues(
-                pm.get_forward_model_configuration()
-            )
-        else:
-            env_pr_fm_step_ = env_pr_fm_step
-
+    def with_plugins(runtime_plugins: ErtRuntimePlugins) -> type[ErtConfig]:
         class ErtConfigWithPlugins(ErtConfig):
             PREINSTALLED_FORWARD_MODEL_STEPS: ClassVar[
                 Mapping[str, ForwardModelStep]
-            ] = preinstalled_fm_steps
-            PREINSTALLED_WORKFLOWS = dict(
-                pm.get_ertscript_workflows().get_workflows()
-            ) | dict(pm.get_legacy_ertscript_workflows().get_workflows())
-            ENV_PR_FM_STEP: ClassVar[dict[str, dict[str, Any]]] = env_pr_fm_step_
-            ACTIVATE_SCRIPT = pm.activate_script()
```

Comment:
The removal of the `activate_script` hook seems a rather independent part of what is being done here. It would simplify understanding what has changed by having a separate commit for that change alone.